workflow/controller/agent.go

```diff
@@ -70,6 +70,43 @@ func assessAgentPodStatus(pod *apiv1.Pod) (wfv1.WorkflowPhase, string) {
 	return newPhase, message
 }
 
+func (woc *wfOperationCtx) secretExists(ctx context.Context, name string) (bool, error) {
+	_, err := woc.controller.kubeclientset.CoreV1().Secrets(woc.wf.Namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if apierr.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (woc *wfOperationCtx) getCertVolumeMount(ctx context.Context, name string) ([]apiv1.Volume, []apiv1.VolumeMount, error) {
+	exists, err := woc.secretExists(ctx, name)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to check if secret %s exists: %v", name, err)
+	}
+	if exists {
+		certVolume := []apiv1.Volume{{
+			Name: name,
+			VolumeSource: apiv1.VolumeSource{
+				Secret: &apiv1.SecretVolumeSource{
+					SecretName: name,
+				},
+			},
+		}}
+
+		certVolumeMount := []apiv1.VolumeMount{{
+			Name:      name,
+			MountPath: "/etc/ssl/certs/ca-certificates",
+			ReadOnly:  true,
+		}}
+
+		return certVolume, certVolumeMount, nil
+	}
+	return nil, nil, nil
```

Comment:
If `!exists` we return `nil` here... have we made sure that wherever `certVolume, certVolumeMount` is used `nil` is an acceptable value? Otherwise we could have panics

Reply:
yes you are right. `nil` is not an acceptable value inside the `[]apiv1.Volume` and `[]apiv1.VolumeMount` so I've added an if else logic to prevent errors.